src/terminal/parser/stateMachine.cpp

```diff
@@ -1270,6 +1437,233 @@ void StateMachine::_EventVt52Param(const wchar_t wch)
+//   2. Ignore Delete characters
+//   3. If we see a ESC, enter the DcsTermination state.
+//   4. Ignore everything else.
+// Arguments:
+// - wch - Character that triggered the event
+// Return Value:
+// - <none>
+void StateMachine::_EventDcsIgnore(const wchar_t wch) noexcept
+{
+    _trace.TraceOnEvent(L"DcsIgnore");
+    if (_isStringTerminator(wch))
+    {
+        _EnterGround();
+    }
+    else if (_isDelete(wch))
+    {
+        _ActionIgnore();
+    }
+    else if (_isCsiParamValue(wch) || _isCsiDelimiter(wch) || _isCsiInvalid(wch) || _isIntermediate(wch) || _isDcsPassThroughValid(wch))
+    {
+        _ActionIgnore();
+    }
+    else if (_isDcsTerminationInitiator(wch))
+    {
+        _EnterDcsTermination();
+    }
+    else
+    {
+        _ActionIgnore();
+    }
+}
+
+// Routine Description:
+// - Processes a character event into an Action that occurs while in the DcsIntermediate state.
+//   Events in this state will:
+//   1. Enter ground on a String terminator
+//   2. Collect intermediate data.
+//   3. Enter DcsPassThrough if we see valid DCS characters
+//   4. If we see a ESC, enter the DcsTermination state.
+//   5. Ignore everything else.
+// Arguments:
+// - wch - Character that triggered the event
+// Return Value:
+// - <none>
+void StateMachine::_EventDcsIntermediate(const wchar_t wch)
+{
+    _trace.TraceOnEvent(L"DcsIntermediate");
+    if (_isStringTerminator(wch))
+    {
+        _EnterGround();
+    }
+    if (_isIntermediate(wch))
+    {
+        _ActionCollect(wch);
+    }
+    else if (_isDcsPassThroughValid(wch))
+    {
+        _ActionDcsPassThrough(wch);
+        _EnterDcsPassThrough();
+    }
+    else if (_isDcsTerminationInitiator(wch))
+    {
+        _EnterDcsTermination();
+    }
+    else
+    {
+        _ActionIgnore();
+    }
+}
+
+// Routine Description:
+// - Processes a character event into an Action that occurs while in the DcsParam state.
+//   Events in this state will:
+//   1. Enter ground on a String terminator
+//   2. Collect DCS parameter data
+//   3. Enter DcsIntermediate if we see an intermediate
+//   4. Enter DcsPassThrough if we see valid DCS characters
+//   5. If we see a ESC, enter the DcsTermination state.
+//   6. Ignore everything else.
+// Arguments:
+// - wch - Character that triggered the event
+// Return Value:
+// - <none>
+void StateMachine::_EventDcsParam(const wchar_t wch)
+{
+    _trace.TraceOnEvent(L"DcsParam");
+    if (_isStringTerminator(wch))
+    {
+        _EnterGround();
+    }
+    else if (_isNumericParamValue(wch) || _isCsiDelimiter(wch))
+    {
+        _ActionParam(wch);
+    }
+    else if (_isIntermediate(wch))
+    {
+        _ActionCollect(wch);
+        _EnterDcsIntermediate();
+    }
+    else if (_isDcsPassThroughValid(wch))
+    {
+        _ActionDcsPassThrough(wch);
+        _EnterDcsPassThrough();
+    }
+    else if (_isDcsTerminationInitiator(wch))
+    {
+        _EnterDcsTermination();
+    }
+    else
+    {
+        _ActionIgnore();

```

Comment:
Again I think we're missing a condition here. The vt100.net state diagram implies that 3A and 3C-3F should enter the ignore state rather than just be ignored.